### ISSUE TITLE
Opening ODBC connection causes panic

### ISSUE THREAD

**Issue by codebybrett** [metaeducation/ren-c#782]
The following lines causes a panic:

    test-db: clean-path %test.s3db

    connection-string: join-all [
        {DSN=SQLite3 Datasource;Database=} (file-to-local test-db)
    ]

    odbc-spec: compose [
        scheme: 'ODBC
        target: (connection-string)
    ]

    connection: open odbc-spec

The panic text:

    PANIC FUNCTION! called (explicitly or by rebPanic() API)
    C Source File --anonymous--, Line 0, Pointer 03A56418
    At evaluator tick: 163667
    Rebol Internal Error
    If you need to file a bug in the issue tracker, please give thorough
    details on how to reproduce the problem:

        https://github.com/metaeducation/ren-c/issues

    Include the following information in the report:

    OS_Free_Mem() mismatched with allocator! Did you mean to use FREE() instead of OS_FREE()?

**Comment by hostilefork** [comment]
This particular error I likely fixed in this commit:

https://github.com/metaeducation/ren-c/commit/e48223ad53979dd8676e793774c7b43c6c88adf5

...but I need to re-set-up a system to do ODBC stuff (as the computer I'd configured to do it died) as it may have atrophied, and we should get tests into the Travis to make sure it stays working...

It's probably ideal if I'm able to test whatever configuration you intend to use.  Is that Sqlite3 on Windows?  Any advice on where to install from, what to click?  (I had all kinds of hassle trying to use SQL Server in Visual Studio before, nothing seemed to work right, even with other clients....so I ultimately used MySQL, which did work).

**Comment by codebybrett** [comment]
Yes, pulling and building latest changes looks to have resolved that error.

Unfortunately more are lurking.  For example, when I tried to create a table I got "unknown ODBC error" but then tried again and found that the table exists.

I have to fess up that I haven't ported my R2 SQLite scripts using Dobeash's SQLite driver to Ren-C and the ODBC driver yet (my main use of SQLite) because it's an ad-hoc personal accounting system. For me that requires a steely motivation to do anything accounting wise, the scripts are complex and depend on some library scripts to be ported; and the hurdle is higher with Ren-c's fluidity.  I do know that I need to redesign my creaking system but I haven't summoned the energy to do that yet. 

Looking at your ODBC test I see that SQLite has a different SQL flavour. E.g it has AUTOINCREMENT but not AUTO_INCREMENT. So different ODBC flavor's are a pain.

I'm using SQLite because I didn't want the hassle of administration for my home system (I haven't tried SQL Server express).  In SQLite's favour is that it is small, but maybe it won't exercise ODBC compatibility as much - don't know.

I use this [SQLite ODBC driver](http://www.ch-werner.de/sqliteodbc/).  I believe I used the provided installers. I installed the 64bit version of the driver for Ren-c.

As an aside, I also installed the 32bit one because I access SQLite from Excel for reporting which also requires the Microsoft Access Database Engine redistributable to get working.

Also the command line shell from [SQLite.org](http://www.sqlite.org/index.html) for looking at the db.

For posterity, it's helpful to know that there are two ODBC Data Source Administrator versions installed in windows - one for 32bit and one for 64bit and not choosing the right one for the job gets confusing.

**Comment by hostilefork** [comment]
I installed SQLite and it seemed to succeed with the create table line I used...no error.  But I'm not totally sure how things correspond to when I was doing the original tests back when, which I made up.

So I'll need literal test.

> the scripts are complex and depend on some library scripts to be ported; and the hurdle is higher with Ren-c's fluidity.

That depends on which side of the fluidity coin you're looking at.  Remember that I want it to be able to emulate Rebol2/Red, and understand any areas where it can't.

Look at the helper [blankify-refinement-args](https://github.com/metaeducation/ren-c/blob/master/src/mezz/mezz-legacy.r#L542), for example.  You could run in a Rebol2 emulation mode and then go one function at a time... converting them to FUNC3.

> For me that requires a steely motivation

Goes faster with help.  I'm getting increased confidence in the decisions, and want to see them tested.  So I have motivation.

I suggest the compatibility route...start from working Rebol2 source, and try to emulate Rebol2 as closely as possible.  R3-Alpha compatibility is essentially irrelevant at this point, so we could start hacking that up...and you could become a practitioner of [the new acrobatics](https://github.com/metaeducation/ren-c/blob/master/src/mezz/base-funcs.r#L586).  :-)

**Comment by hostilefork** [comment]
> Btw, HELP and SOURCE died due to MATCH, 

Was already fixed in a commit after your merge...

> but I haven't looked at why at this point.

Don't waste time looking into bugs like that I can fix quickly, just file an issue.  More productive if you work on things meaningful to you, that keep you interested...

**Comment by hostilefork** [comment]
I got MySQL installed and working, which was an annoyance [because nothing ever works](https://stackoverflow.com/q/49194719/211160)...but the hugeness of the system is a good reminder of why to keep working on trying to think about software another way.

I got the [old ODBC test](https://github.com/metaeducation/ren-c/blob/master/tests/misc/odbc-test.reb) I wrote running again, though the timestamp is buggy and doesn't give back what you put in.  Other than that, ok.

I looked a the sqlite driver for Rebol2 and it doesn't actually look that difficult to port to an extension.  It would probably be about a day's work.  If it would be worth it, I can look into that.
